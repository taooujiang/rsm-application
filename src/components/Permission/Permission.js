import React from 'react'
import PropTypes from 'prop-types'

class Permission extends React.Component {
    render() {
      let {expression} = this.props
      const childrenWithProps = expression ? React.Children.map(this.props.children, child => React.cloneElement(child,{})):null;
      return (childrenWithProps)
    }
}


Permission.propTypes = {
  expression: PropTypes.any
}
Permission.defaultProps = {
  expression:true
}

export default Permission
