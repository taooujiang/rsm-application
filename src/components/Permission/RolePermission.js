import React from 'react'
import PropTypes from 'prop-types'

/**
 * 按角色名称权限进行展示
 * @type {Object}
 */

export default class RolePermission extends React.Component {
  static contextTypes = {
      appConfig: PropTypes.object,
  }
  static propTypes = {
    roleName: PropTypes.string
  }
  static defaultProps = {
    roleName:'normal'
  }

  render() {
      let {roleName} = this.props
      let {appConfig} = this.context
      if(appConfig==undefined){
        //throw new TypeError("RolePermission not has context appConfig props");
        return null
      }else{
        const childrenWithProps = roleName===appConfig.auth.authRole ? React.Children.map(this.props.children, child => React.cloneElement(child,{})):null;
        return (childrenWithProps)
      }
  }
}
