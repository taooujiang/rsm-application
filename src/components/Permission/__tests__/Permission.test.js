
import React from 'react';
import Permission from '../Permission'
import {shallow,mount,render} from 'enzyme';

it('Permission component', () => {
 const tree = shallow(<Permission expression={true}><span>has</span></Permission>);
 expect(tree.contains(<span>has</span>)).toEqual(true);
 expect(tree).toMatchSnapshot();
});


it('Permission component is empty', () => {
 const tree = shallow(<Permission expression={false}><span>has</span></Permission>);
 expect(tree.contains(<span>has</span>)).toEqual(false);
 expect(tree).toMatchSnapshot();
});
