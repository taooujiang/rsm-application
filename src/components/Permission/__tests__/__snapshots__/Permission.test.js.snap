// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Permission component 1`] = `
Array [
  <span
    key=".0"
  />,
]
`;

exports[`Permission component is empty 1`] = `""`;
